app/src/main/java/com/alexbar/layoutbasic/utils/Utils.kt

```diff
@@ -0,0 +1,39 @@
+package com.alexbar.layoutbasic.utils
+
+import androidx.compose.ui.Modifier
+
+inline fun Modifier.conditional(
+    cardType: CardType,
+    ifVisa: Modifier.() -> Modifier,
+    ifMastercard: Modifier.() -> Modifier,
+    ifAmex: Modifier.() -> Modifier,
+    ifUnknown: Modifier.() -> Modifier,
+) : Modifier = when (cardType) {
+    CardType.VISA -> then(ifVisa(Modifier))
+    CardType.MASTERCARD -> then(ifMastercard(Modifier))
+    CardType.AMEX -> then(ifAmex(Modifier))
+    else -> then(ifUnknown(Modifier))
+}
```

Comment:
Could this implementation be considered a conditional modifier? The example in the link you provided implements it with a specific condition and chooses a modifier based on that. In my case, there isn't a direct condition; instead, I wanted to compare the potential types of a credit card and apply a modifier based on that comparison.

Reply:
Yes, your implementation can be considered a conditional modifier.In this case, the condition is based on the comparison of the cardType, and you apply a specific modifier based on that comparison.from my point of view ,  your implementation is a valid and reasonable way to create a conditional modifier based on the comparison of card types.